masonry/src/contexts.rs

```diff
@@ -704,3 +706,146 @@ impl AccessCtx<'_> {
     }
 }
+
+macro_rules! impl_get_raw {
+    ($SomeCtx:tt) => {
+        impl<'s> $SomeCtx<'s> {
+            /// Get a child context and a raw shared reference to a child widget.
+            ///
+            /// The child context can be used to call context methods on behalf of the
+            /// child widget.
+            pub fn get_raw_ref<'a, 'r, Child>(
+                &'a mut self,
+                child: &'a mut WidgetPod<Child>,
+            ) -> RawWrapper<'r, $SomeCtx<'r>, Child>
+            where
+                'a: 'r,
+                's: 'r,
+            {
+                #[allow(clippy::needless_update)]
+                let child_ctx = $SomeCtx {
+                    widget_state: &mut child.state,
+                    global_state: self.global_state,
+                    ..*self
+                };
+                RawWrapper {
+                    ctx: child_ctx,
+                    widget: &child.inner,
+                }
+            }
+
+            /// Get a raw mutable reference to a child widget.
+            ///
+            /// See documentation for [`AllowRawMut`] for more details.
+            pub fn get_raw_mut<'a, 'r, Child>(
+                &'a mut self,
+                child: &'a mut WidgetPod<Child>,
+            ) -> RawWrapperMut<'r, $SomeCtx<'r>, Child>
+            where
+                Child: AllowRawMut,
+                'a: 'r,
+                's: 'r,
+            {
+                #[allow(clippy::needless_update)]
+                let child_ctx = $SomeCtx {
+                    widget_state: &mut child.state,
+                    global_state: self.global_state,
+                    ..*self
+                };
+                RawWrapperMut {
+                    parent_widget_state: &mut self.widget_state,
+                    ctx: child_ctx,
+                    widget: &mut child.inner,
+                }
+            }
+        }
+    };
+}
+
+impl_get_raw!(EventCtx);
+impl_get_raw!(LifeCycleCtx);
+impl_get_raw!(LayoutCtx);
+
+impl<'s> AccessCtx<'s> {
+    pub fn get_raw_ref<'a, 'r, Child>(
+        &'a mut self,
+        child: &'a WidgetPod<Child>,
+    ) -> RawWrapper<'r, AccessCtx<'r>, Child>
+    where
+        'a: 'r,
+        's: 'r,
+    {
+        let child_ctx = AccessCtx {
+            widget_state: &child.state,
+            global_state: self.global_state,
+            tree_update: self.tree_update,
+            // TODO - This doesn't make sense. NodeBuilder should probably be split
+            // out from AccessCtx.
+            current_node: NodeBuilder::default(),
+            rebuild_all: self.rebuild_all,
+            scale_factor: self.scale_factor,
+        };
+        RawWrapper {
+            ctx: child_ctx,
+            widget: &child.inner,
+        }
+    }
+}
+
+// --- MARK: WRAPPER
+
+pub struct RawWrapper<'a, Ctx, W> {
+    pub(crate) ctx: Ctx,
+    pub(crate) widget: &'a W,
+}
+
+pub struct RawWrapperMut<'a, Ctx: Context, W> {
```

Comment:
The warning about `Context` being private is unfortunate. I think I'd prefer to make `Context` a sealed trait rather than a private one, as the resultant docs of `RawWrapperMut` are quite poor.

Reply:
I've done that in the last version, though I'm not super confident in my code.